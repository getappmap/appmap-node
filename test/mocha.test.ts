import { integrationTest, readAppmaps, runAppmapNode } from "./helpers";

integrationTest("mapping Mocha tests", () => {
  expect(runAppmapNode("yarn", "mocha").status).toBe(1);
  expect(readAppmaps()).toMatchSnapshot();
});
