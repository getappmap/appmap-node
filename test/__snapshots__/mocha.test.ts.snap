// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapping Mocha tests 1`] = `
{
  "./tmp/appmap/mocha/exception handling/intentionally throws.appmap.json": {
    "classMap": [
      {
        "children": [
          {
            "children": [
              {
                "location": "./test.js:18",
                "name": "errorOut",
                "static": true,
                "type": "function",
              },
            ],
            "name": "test",
            "type": "class",
          },
        ],
        "name": "test",
        "type": "package",
      },
    ],
    "events": [
      {
        "defined_class": "test",
        "event": "call",
        "id": 1,
        "lineno": 18,
        "method_id": "errorOut",
        "parameters": [],
        "path": "./test.js",
        "static": true,
        "thread_id": 0,
      },
      {
        "elapsed": 31.337,
        "event": "return",
        "exceptions": [
          {
            "class": "TestError",
            "message": "test error",
            "object_id": 2,
          },
        ],
        "id": 2,
        "parent_id": 1,
        "thread_id": 0,
      },
    ],
    "metadata": {
      "app": "mocha-appmap-node-test",
      "client": {
        "name": "appmap-node",
        "url": "https://github.com/getappmap/appmap-node",
        "version": "test node-appmap version",
      },
      "exception": {
        "class": "TestError",
        "message": "test error",
      },
      "language": {
        "engine": "Node.js",
        "name": "javascript",
        "version": "test node version",
      },
      "name": "exception handling intentionally throws",
      "recorder": {
        "name": "mocha",
        "type": "tests",
      },
      "test_status": "failed",
    },
    "version": "1.12",
  },
  "./tmp/appmap/mocha/multiply/calculates power correctly.appmap.json": {
    "classMap": [
      {
        "children": [
          {
            "children": [
              {
                "location": "./calc.js:5",
                "name": "power",
                "static": true,
                "type": "function",
              },
            ],
            "name": "calc",
            "type": "class",
          },
        ],
        "name": "calc",
        "type": "package",
      },
    ],
    "events": [
      {
        "defined_class": "calc",
        "event": "call",
        "id": 1,
        "lineno": 5,
        "method_id": "power",
        "parameters": [
          {
            "class": "Number",
            "name": "x",
            "value": "3",
          },
          {
            "class": "Number",
            "name": "y",
            "value": "2",
          },
        ],
        "path": "./calc.js",
        "static": true,
        "thread_id": 0,
      },
      {
        "elapsed": 31.337,
        "event": "return",
        "id": 2,
        "parent_id": 1,
        "return_value": {
          "class": "Number",
          "value": "8",
        },
        "thread_id": 0,
      },
    ],
    "metadata": {
      "app": "mocha-appmap-node-test",
      "client": {
        "name": "appmap-node",
        "url": "https://github.com/getappmap/appmap-node",
        "version": "test node-appmap version",
      },
      "exception": {
        "class": "AssertionError",
        "message": "8 == 9",
      },
      "language": {
        "engine": "Node.js",
        "name": "javascript",
        "version": "test node version",
      },
      "name": "multiply calculates power correctly",
      "recorder": {
        "name": "mocha",
        "type": "tests",
      },
      "test_status": "failed",
    },
    "version": "1.12",
  },
  "./tmp/appmap/mocha/multiply/multiplies numbers correctly.appmap.json": {
    "classMap": [
      {
        "children": [
          {
            "children": [
              {
                "location": "./calc.js:1",
                "name": "multiply",
                "static": true,
                "type": "function",
              },
            ],
            "name": "calc",
            "type": "class",
          },
        ],
        "name": "calc",
        "type": "package",
      },
    ],
    "events": [
      {
        "defined_class": "calc",
        "event": "call",
        "id": 1,
        "lineno": 1,
        "method_id": "multiply",
        "parameters": [
          {
            "class": "Number",
            "name": "x",
            "value": "2",
          },
          {
            "class": "Number",
            "name": "y",
            "value": "5",
          },
        ],
        "path": "./calc.js",
        "static": true,
        "thread_id": 0,
      },
      {
        "elapsed": 31.337,
        "event": "return",
        "id": 2,
        "parent_id": 1,
        "return_value": {
          "class": "Number",
          "value": "10",
        },
        "thread_id": 0,
      },
    ],
    "metadata": {
      "app": "mocha-appmap-node-test",
      "client": {
        "name": "appmap-node",
        "url": "https://github.com/getappmap/appmap-node",
        "version": "test node-appmap version",
      },
      "language": {
        "engine": "Node.js",
        "name": "javascript",
        "version": "test node version",
      },
      "name": "multiply multiplies numbers correctly",
      "recorder": {
        "name": "mocha",
        "type": "tests",
      },
      "test_status": "succeeded",
    },
    "version": "1.12",
  },
}
`;
